### ISSUE TITLE
Externalization of jHipster's Angular code as an autonomous NPM dependency ?

### ISSUE THREAD

**Issue by astik** [jhipster/generator-jhipster#2886]
Hi all,
Just wondering about the possibility of externalizing JS code (generate by jHipster generator) as a reusable NPM dependency.
For example, with the jHipster generator, I am able to generate all the needed code to start working with a clean admin interface (CRUD, metrics, log, ...). Authentication is perfectly managed on back end and frontend.
What if I need to write another frontend for my API (like another frontend more business oriented or a more restricted mobile app) ? I will reuse the authentication part, maybe as identical as the jHipster generated files. So that, it will be perfect to be able to rely on a simple NPM dependency to retrieve all authentication code (token management, retrieval, storage, refresh, ...).
In other words, would it be possible to externalize components/auth, components/user, components/interceptor in a external NPM dependencies ?
Of course, I'm assuming that each authentication mechanism will have its own package (token, oauth, ...).
Thanks in advance.

**Comment by deepu105** [comment]
Technically possible but its not in our roadmap atleast for now so I dont think we will have any time for this.
And another reason we might not go that direction is that we want to give our users full control of the source code and that is possible only when you generate it as part of your code, making these external bower/npm libs mean users canot customize.
